myFitnessTrainer/src/screens/WorkoutScreen.js

```diff
@@ -1,108 +1,184 @@
+    const dailyExerciseSet = route.params;
+
+    // adds additional card at end
+    const exerciseDeck = dailyExerciseSet.concat([{ name: "Finished Card" }]);
+
+    // currentExerciseStats represents the stats of the current exercise on the card
+    // currentWorkoutStats represents the the combined stats of every card in the daily workout
+    const workoutLength = dailyExerciseSet.length;
+    const [currentExerciseStats, setCurrentExerciseStats] = useState("");
     const [exerciseIndex, setExerciseIndex] = useState(0);
-    const [progress, setProgress] = useState(0/workoutLength);
-    const height = Dimensions.get('window').height;
+    const [progress, setProgress] = useState(0 / workoutLength);
+    const height = Dimensions.get("window").height;
+
+    // when exerciseIndex state changes, then we will fetch the new default exercise stats from DB
+    // currentExerciseStats state will reflect the stats of the current exrercise shown on the card
+    useEffect(() => {
+        // the last card is not a valid exercise
+        if (exerciseIndex < workoutLength - 1) {
+            getAllDocuments("exercise_stats").then((retrievedDocuments) => {
+                const exerciseStatRef =
+                    dailyExerciseSet[exerciseIndex]["default_exercise_stat_id"];
+                const targetStatId = exerciseStatRef["path"].split("/")[1];
+
+                const filteredStats = Object.keys(retrievedDocuments).filter(
+                    (statId) => statId === targetStatId
+                );
+                setCurrentExerciseStats(retrievedDocuments[filteredStats]);
+            });
+        }
+    }, [exerciseIndex]);
 
     const progressFooter = (
         <View style={styles.bottomProgressContainer}>
             <View style={styles.progressBarContainer}>
                 <Progress.Bar
                     progress={progress}
                     pas
-                    width={width * .8}
+                    width={width * 0.8}
                     height={20}
-                    color='#4682B4'
+                    color="#4682B4"
                 />
             </View>
-            <Text style={styles.progressText}>{exerciseIndex} of {workoutLength} exercises completed</Text>
+            <Text style={styles.progressText}>
+                {exerciseIndex} of {workoutLength} exercises completed
+            </Text>
         </View>
     );
 
-    const renderCarouselItem = ({index}) => {
+    const submitStatsToDB = async () => {
+        const exercise = dailyExerciseSet[exerciseIndex];
+        const exerciseId = await getDocumentId(
+            "exercise_library",
+            "name",
+            exercise["name"]
+        );
+
+        const userId = await getDocumentId(
+            "users",
+            "email",
+            auth.currentUser.email
+        );
+        const postObject = {
```

Comment:
In our project plan, we had the following attributes for ExerciseHistory:
<img width="224" alt="image" src="https://github.com/osu-cs467-myFitnessTrainer/myFitnessTrainer/assets/21234456/f74975f1-71c0-4357-a40e-0207d7ad1209">

If we have the postObject reflecting this, it'll simplify what we upload to exercise_history

Reply:
Fixed to reflect what is shown on the report. Thanks.